fair_mango/metrics/metrics.py

```diff
@@ -549,17 +586,20 @@ def summary(self) -> dict:
                     else:
                         self.result[target][self.label1] = key[1]
                         self.result[target][self.label2] = key[0]
+
         return self.result
 
-    def rank(self, pr_to_unp: bool = True) -> dict:
-        self.pr_to_unp = pr_to_unp
-        if self.result is None:
-            self.summary()
+    def rank(self) -> dict:
         result: dict = {}
         self.ranking = {}
+
         for target in self.data.real_target:
             result.setdefault(target, {})
             self.ranking.setdefault(target, {})
+
+        if self.results is None:
+            self.results = self._compute()
+
         for key, values in self.results.items():
             if isinstance(values, float):
                 values = [values]
```

Comment:
this check can be useless depending on the return type of `def difference(...) -> ...`. If `difference` returns only a dict of list, then you are never gonna enter this condition

Reply:
I checked the `.difference()` function and it returns `dict[tuple, float | np.ndarray[float]]. Which means that the check I am doing is needed.
PS: I added the typing to the returned value of `.difference()` and `.ratio()` functions.

Reply:
in what case does it return a float ?

Reply:
do you think it would be better to make difference() always output a numpy array to clean up higher level code?

Reply:
Done. I made sure the functions `difference` and `ratio` always return a 1D numpy array.

Reply:
FYI, you use `MAINT` too often I think. `MAINT` is meant (ahah) for :

> MAINT: maintenance commit (refactoring, typos, etc.)

When you improve your code you can use `ENH` instead

Reply:
ratio and difference are way too similar to be two different functions. You can try to use a generator to factorize

Reply:
Since the code does the same thing in another way I consider refactoring and use `MAINT` tag. I try to use `ENH` when I add something new.
Should I use `ENH` for refactoring commits that optimise the code too?

Reply:
hummm good question, I was seing refactorings as non-functional changes, but in a sense you are right. The higher level interface did not change.
Well done